File: app/app/src/main/java/com/hobbymeetingapp/app/Utils/NonScrollRecyclerView.java
package com.hobbymeetingapp.app.Utils;

import android.content.Context;
import android.util.AttributeSet;
import androidx.recyclerview.widget.RecyclerView;

public class NonScrollRecyclerView extends RecyclerView {

    public NonScrollRecyclerView(Context context) {
        super(context);
    }

    public NonScrollRecyclerView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public NonScrollRecyclerView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    @Override
    public void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        int heightMeasureSpec_custom = MeasureSpec.makeMeasureSpec(
                Integer.MAX_VALUE >> 1, MeasureSpec.AT_MOST);
        super.onMeasure(widthMeasureSpec, heightMeasureSpec_custom);
    }
}

File: app/app/src/main/java/com/cabal/app/Utils/NonScrollRecyclerView.kt
package com.cabal.app.Utils

import androidx.recyclerview.widget.RecyclerView
import android.content.Context
import android.util.AttributeSet

class NonScrollRecyclerView: RecyclerView {

    constructor(context: Context): super(context)
    constructor(context: Context, attributeSet: AttributeSet) : super(context, attributeSet)
    constructor(context: Context, attributeSet: AttributeSet, defStyle: Int): super(context, attributeSet, defStyle)

    override fun onMeasure(widthSpec: Int, heightSpec: Int) {
        val heightmeasuerspecCustom = MeasureSpec.makeMeasureSpec(Int.MAX_VALUE shr 1, MeasureSpec.AT_MOST)
        super.onMeasure(widthSpec, heightmeasuerspecCustom)
    }
}